robot_sort/robot_sort.py

```diff
@@ -96,8 +98,32 @@ def sort(self):
         """
         Sort the robot's list.
         """
-        # Fill this out
-        pass
+        if not can_move_right(self) and not light_is_on(self):
+            return None
+
+        elif not can_move_right(self) and light_is_on(self):
+            while can_move_left(self):
+                move_left(self)
+            set_light_off(self)
+            sort(self)
+        
+        elif can_move_right(self) is True:
+            swap_item(self)
+            move_right(self)
+
+            if compare_item(self) == 1:
+                swap_item(self)
+                move_left(self)
+                swap_item(self)
+                move_right(self)
+                set_light_on(self)
+                sort(self)
+
+            else:
+                move_left(self)
+                swap_item(self)
+                move_right(self)
+                sort(self)
 
```

Comment:
very cool implementation; as you noticed too, it is a little slow. You should attempt to refactor the code and try to DRY it up a bit.